docs/sql-ref-functions.md

```diff
@@ -27,12 +27,15 @@ Built-in functions are commonly used routines that Spark SQL predefines and a co
 Spark SQL has some categories of frequently-used built-in functions for aggregtion, arrays/maps, date/timestamp, and JSON data.
 This subsection presents the usages and descriptions of these functions.
 
- * [Aggregate Functions](sql-ref-functions-builtin.html#aggregate-functions)
- * [Window Functions](sql-ref-functions-builtin.html#window-functions)
+#### Scalar Functions
  * [Array Functions](sql-ref-functions-builtin.html#array-functions)
  * [Map Functions](sql-ref-functions-builtin.html#map-functions)
  * [Date and Timestamp Functions](sql-ref-functions-builtin.html#date-and-timestamp-functions)
  * [JSON Functions](sql-ref-functions-builtin.html#json-functions)
+ * [Window Functions](sql-ref-functions-builtin.html#window-functions)
```

Comment:
> OK... Seems Window functions still should be put in the Scalar function category. I will double check tomorrow.

Why do you think so? Putting this in the scalar category still looks weird to me.
The other parts look ok. Also, could you file a jira, just in case?

Reply:
I found this https://issues.apache.org/jira/browse/SPARK-29458. I will use this jira.

Reply:
By definition, scalar function returns a single value per row, while aggregate function returns a single value on a group of rows. Since window function returns a value per row, seems more suitable to put into scalar function category.

Reply:
I think vector input & output rather than scalar input & output. WDYT? @HyukjinKwon

Reply:
This is how snowflake categorizes the functions: https://docs.snowflake.com/en/sql-reference-functions.html. 
Snowflake lists window function in parallel with scalar function and aggregate function, but it also says "Window Functions — subset of aggregate functions that can operate on a subset of rows."

I guess it's safe to use your proposal
```
#### Aggregate-like Functions
Aggregate Functions
Window Functions
```

Reply:
Yea, the last one looks fine. I think we can improve the structure later if necessary.

Reply:
Sorry, I am a little confused...  you prefer to put window function in scalar function, or aggregate-like function?

Reply:
`aggregate-like function` one.

Reply:
OK... let me change it now